packages/braid-design-system/src/lib/components/Autosuggest/reverseMatches.test.ts

```diff
@@ -0,0 +1,58 @@
+import { type Match, reverseMatches } from './reverseMatches';
+
+describe('reverseMatches', () => {
+  it('should return intervals for non-matching parts of the suggestion', () => {
+    const suggestion = 'Apples etc';
+    const matches: Match[] = [
+      [2, 4],
+      [6, 8],
+    ];
+    const expected: Match[] = [
+      [0, 2],
+      [4, 6],
+      [8, 10],
+    ];
+
+    expect(reverseMatches(suggestion, matches)).toEqual(expected);
+  });
+
+  it('should handle no matches', () => {
+    const suggestion = 'Apple';
+    const matches: Match[] = [];
+    const expected: Match[] = [[0, 5]];
+
+    expect(reverseMatches(suggestion, matches)).toEqual(expected);
+  });
+
+  it('should handle matches that cover the entire suggestion', () => {
+    const suggestion = 'Apple';
+    const matches: Match[] = [[0, 5]];
+    const expected: Match[] = [];
+
+    expect(reverseMatches(suggestion, matches)).toEqual(expected);
+  });
+
+  it('should handle matches at the start of the suggestion', () => {
+    const suggestion = 'Apple';
+    const matches: Match[] = [[0, 2]];
+    const expected: Match[] = [[2, 5]];
```

Comment:
Looking at these numbers, im not sure I agree. If we match `A P P` (e.g. `[0,2]`) and we reverse that, wouldnt we be expecting `L E` (e.g. `[3, 4]`)?